src/app/components/chat/room-details-modal/form/form.ts

```diff
@@ -0,0 +1,20 @@
+import { Component, Emit } from 'vue-property-decorator';
+import { BaseForm } from '../../../../../_common/form-vue/form.service';
+import { ChatRoom } from '../../room';
+
+export type FormModel = {
+	title: string;
+};
+
+@Component({})
+export default class FormRoomDetails extends BaseForm<FormModel> {
+	modelClass = ChatRoom;
+
+	@Emit('submit')
+	emitSubmit(_title: FormModel) {}
+
+	onSubmit() {
+		const submit: FormModel = { title: this.formModel.title };
```

Comment:
Typescript should already have the typing information - we shouldn't need to specifically describe a type here.

Reply:
actually, I think we should just emit the entire `formModel` object instead of extracting the title like this. The function that triggers from the emit can then extract any information it needs from the event and do what it needs to with it.